### PR TITLE
user authentication

### PR DESCRIPTION
**Pull request by emmygozi**
#### What does this PR do?

User can signup and login to the application
#### Description of Task to be completed?

- setup webpack 
- setup boilerplate code for redux
- create a project structure
- user can signup on application
- user can login to the application
- Add toast notification for success and error messages
#### How should this be manually tested?

clone application `git clone https://github.com/the-sage-org/How-to-FE.git`
run `yarn install` to install project dependencies
run `yarn dev` to start development server
#### Any background context you want to provide?

N/A
#### What are the relevant pivotal tracker stories?

N/A
#### Screenshots (if appropriate)

image
image
image

### CODE REVIEW

File: src/actions/action.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
+export const LOGIN_FAILURE = 'LOGIN_FAILURE';
+
+export const REGISTER_SUCCESS = 'REGISTER_SUCCESS';
+export const REGISTER_FAILURE = 'REGISTER_FAILURE';
```
Comment (SorinC6) [inline src/actions/action.js:5]: For a improvement I will add another  action type like `FETCHING_START ` so you you can keep track when your action request as started so you can implement a easy Spinner maybe

File: src/component/loginForm.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  constructor() {
+    super();
+    this.state = {
+      email: "",
+      password: ""
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
```
Comment (SorinC6) [inline src/component/loginForm.jsx:17]: Try to use the arrow syntax more so you don't have to bind function anymore in React

**Review by SorinC6 — COMMENTED** [review]
- Emmanuel, nice progress on the project so far.
- You have properly implemented React with Redux and created clear structure of components. You also added action creators with reducers.
- I really liked that you had time to stye the app
- For future improvements your git workflow could be improved with more  detailed and meaningful  commits messages for every feature of the app.
- I know its soon but think about deploying your app. I can see that the backend its on heroku so there should't be any problems 👍

**Comment by emmygozi** [comment]
Thanks for your feedback, Sorin. Would implement feedback on next PR.

**Pull request merged**